packages/components/src/components/tooltip/Tooltip.tsx

```diff
@@ -1,176 +1,155 @@
-  onHintClose?: () => void;
-  placement: 'top' | 'bottom' | 'left' | 'right';
-  type?: 'hint' | 'tooltip';
-  /** if true, the tooltip should be displayed */
-  visible?: boolean;
-  /** Optional CSS class name for wrapping span element  */
-  wrapperClassName?: string;
-}
-
-export const Tooltip: React.FC<TooltipProps> = ({
-  children,
-  closeLabel,
-  description,
-  displayTrigger,
-  id,
-  onHintClose,
-  placement,
-  type,
-  visible,
-  className,
-  wrapperClassName,
-  hoverDelay = 0,
-  hoverTimeout = 100,
-  ...otherProps
-}) => {
-  const arrowRef = React.useRef(null);
-  const [isOpen, setIsOpen] = useState(visible);
-
-  React.useEffect(() => {
-    setIsOpen(visible)
-  }, [visible])
-  
-  const { refs, floatingStyles, context } = useFloating({
-    middleware: [
-      arrow({
-        element: arrowRef,
-      }),
-      flip(),
-      offset(type === 'hint' ? 15 : 9),
-    ],
-    open: isOpen,
-    onOpenChange: setIsOpen,
-    placement: placement ?? 'top',
-    whileElementsMounted: autoUpdate,
-  });
-
-  const dismiss = useDismiss(context, {
-    enabled: displayTrigger === 'click',
-  });
-  
-  const hover = useHover(context, {
-    delay: {
-      close: hoverTimeout,
-      open: hoverDelay,
-    },
-    enabled: displayTrigger === 'hover'
-  });
-
-  const { getReferenceProps, getFloatingProps } = useInteractions([
-    dismiss,
-    hover,
-  ])
-
-  return <>
-    { /** The element tooltip is wrapped around. */ }
-    <div
-      className={ clsx('tk-tooltip__wrapper', wrapperClassName) }
-      onClick={() => {
-        displayTrigger === 'click' && setIsOpen(!isOpen)
-      }}
-      ref={ refs.setReference }
-      { ...getReferenceProps() }
-    >
-      { children }
-    </div>
-
-    { /** The tooltip. */ }
-    { isOpen && <div
-      className={clsx(
-        `${type === 'tooltip' ? 'tk-tooltip' : 'tk-hint'}`,
-        className
-      )}
-      id={id}
-      role="tooltip"
-      ref={refs.setFloating}
-      style={floatingStyles}
-      {...otherProps}
-      { ...getFloatingProps() }
-    >
-      <span className="tk-hint__description">{description}</span>
-      {type === 'hint' && <div className="tk-hint__footer">
-        {closeLabel ? (
-          <span
-            className="tk-hint__close"
-            onClick={() => {
-              setIsOpen(false)
-              onHintClose && onHintClose()
-            }}
-            onKeyDown={(event) => {
-              if(event.key === 'Enter') {
-                onHintClose && onHintClose()
-                setIsOpen(false)
-              }
-            }}
-            tabIndex={0}
-          >
-            {closeLabel}
-          </span>
-        ) : null}
-      </div>
-      }
-      { type === 'hint' && <FloatingArrow
-        context={context}
-        fill="#2b2d32"
-        ref={arrowRef}
-      /> }
-    </div>
-    }
-  </>
-};
-
-Tooltip.defaultProps = {
-  type: 'hint',
-};
-
-Tooltip.propTypes = {
```

Comment:
@antoinerollindev @aurelie-forzani @benoit-sy do you get usage from prop-styles package? I feel like TypeScript should be enough and it's a burden having both maintained.

Reply:
@axeleriksson147 The only use-case where PropTypes still makes sense over pure TypeScript is for when dynamic data are passed to components.
So we could maybe plan for a global refactor PR where we ditch entirely prop-types in favour or TS types/interfaces (that most of the time we already have)

Reply:
Yeah, and in that case it's about getting an "easier" error message but the component will error either way I believe.